File: 2022-06-craft-conf-languages/examples/files.c
#include <stdio.h>

int
save_file(const char* filename)
{
    FILE* fp = fopen(filename, "w");
    if (fp == NULL) goto error0;

    int rc = fputs("lots of interesting data", fp);
    if (rc < 0) goto error1;

    fclose(fp);
    return 0;

error1:
    fclose(fp);
error0:
    return -1;
}

int
main(void)
{
    save_file("out.dat");
}

File: 2022-06-craft-conf-languages/examples/files.rs
use std::fs::File;
use std::io::Write;

fn save_file(filename: &str) -> Result<(), std::io::Error> {
    let mut fp = File::create(filename)?;
    write!(fp, "lots of interesting data")?;
    Ok(())
}

fn main() {
    save_file("out.dat").unwrap();
}
